conda/core/index.py

```diff
@@ -114,7 +115,7 @@ def fetch_index(channel_urls, use_cache=False, index=None):
 
     if index is None:
         index = {}
-    for _, repodata in repodatas:
+    for _, repodata in reversed(repodatas):
```

Comment:
This is a major change in logic, but only affects the contents of `index` if there's a hash collision.  When there *is* a hash collision, this is actually the correct logic.